zerver/lib/message.py

```diff
@@ -634,6 +635,13 @@ def bulk_access_messages(user_profile: UserProfile, messages: Sequence[Message])
             filtered_messages.append(message)
     return filtered_messages
 
+def bulk_access_messages_expect_archive_usermessage(
```

Comment:
Given that `bulk_access_messages_expect_usermessage` is used only in one plane in our codebase, that being the "handle remove notification" codepath, I think it might be better to keep things in that one function, renaming it to something explicit like `bulk_access_messages_expect_usermessage_including_deleted` (super long name, but idk a better one right now :disappointed: )

Reply:
Also, if it's in one function can we perhaps `UNION` the two queries to make things a bit more efficient?